src/other-reprs.md

```diff
@@ -112,26 +112,43 @@ assert_eq!(16, size_of::<MyReprOption<&u16>>());
 
 This optimization still applies to fieldless enums with an explicit `repr(u*)`, `repr(i*)`, or `repr(C)`.
 
-## repr(packed)
+## repr(packed(n))
 
-`repr(packed)` forces Rust to strip any padding, and only align the type to a
-byte. This may improve the memory footprint, but will likely have other negative
-side-effects.
+`repr(packed(n))` forces the Rust compiler to strip any padding between fields, and align the type, hence,
```

Comment:
This still isn't quite correct, as it may not necessarily remove padding between fields.

Reply:
What do you mean by that? Isn't `packed` exactly there to strip any padding between struct members? @ehuss

Reply:
For example, in the following, there is still 1 byte of padding between the first and second field:

```rust
#[repr(packed(2))]
struct Foo {
    f1: u8,
    f2: u32,
}
```

Reply:
Ohh, I learned something! Thanks! I thought that only the beginning of the struct gets aligned to n and all other properties have no padding.

Reply:
@ehuss What's the difference between align(2) and packed(2) then, if packed doesn't have zero padding between the fields for N > 1?

Reply:
Sorry, I'm not sure I understand the question.  `align` and `packed` do opposite things.  `align` can increase the alignment of the type itself, and `packed` will lower it. In addition, `packed` can also change the padding between fields of a struct, as it will set the alignment of each field to the smaller of the specified alignment and the normal alignment of that field's type.  `align` does not change the alignment of each field, so won't change the field padding.

So, with the example above, `Foo` has 2-byte alignment, with 1 byte of padding between `f1` and `f2`.
If you use `align(2)`, then `Foo` has 4-byte alignment, with 3 bytes of padding between `f1` and `f2`, which is the same if you don't specify `align` at all.